packages/config/src/api/site_info.ts

```diff
@@ -105,12 +103,28 @@ const getSite = async function (api: NetlifyAPI, siteId: string, siteFeatureFlag
   }
 }
 
-const getAccounts = async function (api: NetlifyAPI) {
+export type MinimalAccount = {
+  id: string
+  name: string
+  slug: string
+  default: boolean
+  team_logo_url: string | null
+  on_pro_trial: boolean
+  organization_id: string | null
+  type_name: string
+  type_slug: string
+  members_count: number
+}
+
+const getAccounts = async function (api: NetlifyAPI): Promise<MinimalAccount[]> {
   try {
-    const accounts = await (api as any).listAccountsForUser()
-    return Array.isArray(accounts) ? accounts : []
+    const accounts = (await api.listAccountsForUser(
+      // @ts-expect-error(ndhoule): This is an unpublished, internal querystring parameter
+      { minimal: 'true' },
+    )) as MinimalAccount[] | null
+    return Array.isArray(accounts) ? (accounts as MinimalAccount[]) : []
```

Comment:
🤔 why is this type assertion necessary? does `Array.isArray` not refine the union?

Reply:
Sadly, it does not.